lib/core/public/run.js

```diff
@@ -112,8 +112,9 @@ axe.run = function (context, options, callback) {
 		});
 	}
 
-	axe._runRules(context, options, function (rawResults) {
+	axe._runRules(context, options, function (rawResults, cleanup) {
 		let respond = function (results) {
+			cleanup();
```

Comment:
Can you could add a comment about why you had to run `cleanup` before `resolve` here since it differs from above?